Redesigned Navigation Bar for Improved UI Clarity and Accessibility : UI changes

sahilsobhani opened the pull request:
###  Key Changes

This PR updates the mobile navigation bar for a cleaner, more readable, and accessible design. The new layout improves usability by replacing icon-only links with labeled buttons, offering better clarity for first-time users and improved touch targets on mobile devices.

## How it looks
| Old | New |
|-----|-----|
| <img width="254" height="562" alt="Screenshot 2025-07-15 013548" src="https://github.com/user-attachments/assets/7462c1a0-0760-4066-a4bc-057e33857e27" /> | <img width="256" height="557" alt="Screenshot 2025-07-15 013528" src="https://github.com/user-attachments/assets/97d8273b-6f0f-4480-8e71-0731a3ea61c7" />

## Checklist
 - Visual changes tested on mobile and desktop
 - Added Nuxt/icons lib
 - Fully responsive and consistent with overall design system
 - Accessibility improved for screen readers and users with visual impairments
 - Visuals are improved on large screen devices

<img width="1265" height="625" alt="image" src="https://github.com/user-attachments/assets/95c4d444-3db8-4118-9155-2de099cbc13c" />
 
### Any changes needed?
Please advise if any changes are required from my end!

tomhaerter commented:
@sahilsobhani thank you for the PR!

a few small things

- you added `uppercase` to the marketing button class, please remove that class


1) on mobile the navigation seems to jump to the right

![telegram-cloud-photo-size-2-5451727312119790687-y](https://github.com/user-attachments/assets/ac6cb0bf-a486-4159-883b-58ed01a27c02)

if you can fix these, i am happy to review again and merge it all looks good

sahilsobhani commented:
> @sahilsobhani thank you for the PR!
> 
> a few small things
> 
> 1. you added `uppercase` to the marketing button class, please remove that class
> 2. on mobile the navigation seems to jump to the right
> 
> ![telegram-cloud-photo-size-2-5451727312119790687-y](https://private-user-images.githubusercontent.com/16882922/466202825-ac6cb0bf-a486-4159-883b-58ed01a27c02.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1MjkxMTUsIm5iZiI6MTc1MjUyODgxNSwicGF0aCI6Ii8xNjg4MjkyMi80NjYyMDI4MjUtYWM2Y2IwYmYtYTQ4Ni00MTU5LTg4M2ItNThlZDAxYTI3YzAyLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDIxMzMzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3YzkwOWQ0ZmI4OTkzODJlOTAzNGM1YzNmOWQzZGM4MzE1NTU3NWNlYTViOGNlMmMyOGQwZDk2Y2I0MzIyNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZohRy8Awsm0UaaF3-eQCjU4M8VTW_1hIv4nu3S8E150)
> 
> if you can fix these, i am happy to review again and merge it all looks good

### Made the requested changes

Hi! Thanks for the feedback.
I've made the requested changes:

- Removed the uppercase class from the marketing button.
- Fixed the mobile navigation jump issue.

Let me know if there's anything else you'd like me to tweak - happy to iterate!

tomhaerter reviewed (changes requested):
@sahilsobhani amazing! could you remove the package-lock.json? we are using bun for dependency management and thus no package-lock.json

sahilsobhani commented:
> @sahilsobhani amazing! all but one thing looks good now! could you remove the package-lock.json? we are using bun for dependency management and thus no package-lock.json

#### Removed package-lock.json
